bigml/dataset.py

```diff
@@ -0,0 +1,189 @@
+    return new_fields
+
+
+class Dataset:
+    """Local representation of a BigML Dataset. It can store a sample of
+    data whose fields are a subset of the ones defined in the fields
+    attribute.
+    """
+    def __init__(self, dataset, api=None, cache_get=None):
+        if use_cache(cache_get):
+            self.__dict__ = load(get_dataset_id(dataset), cache_get)
+            return
+
+        self.resource_id = None
+        self.rows = None
+        self.origin_dataset = None
+        self.in_fields = None
+        self.out_fields = None
+        self.description = None
+        self.locale = None
+        self.api = get_api_connection(api)
+        self.transformations = None
+
+        # retrieving dataset information from
+        self.resource_id, dataset = get_resource_dict( \
+            dataset, "dataset", api=self.api, no_check_fields=False)
+
+        if 'object' in dataset and isinstance(dataset['object'], dict):
+            dataset = dataset['object']
+
+        if 'fields' in dataset and isinstance(dataset['fields'], dict):
+            status = get_status(dataset)
+            if 'code' in status and status['code'] == FINISHED:
+                self.out_fields = Fields(dataset)
+                self.out_header_names, _ = sorted_headers(self.out_fields)
+                self.description = dataset["description"]
+                self.locale = dataset.get('locale', DEFAULT_LOCALE)
+                self.rows = dataset.get("rows", 0)
+                # we extract the generators and names from the "output_fields"
+                new_fields = get_new_fields(dataset.get("output_fields", []))
+                origin_dataset = dataset.get("origin_dataset")
+                if origin_dataset:
+                    self.add_transformations(origin_dataset, new_fields)
+                    self.in_header_names, self.in_header_ids = sorted_headers(
+                        self.in_fields)
+
+    def add_transformations(self, origin_dataset, new_fields):
+        """Adds a new transformation where the new fields provided are
+        defined
+        """
+        origin_dataset = self.api.get_dataset(origin_dataset)
+        self.origin_dataset = Dataset(origin_dataset)
+        self.in_fields = self.origin_dataset.out_fields
+        if new_fields:
+            self.transformations = new_fields
+
+    def get_sample(self, rows_number=32):
+        """Gets a sample of data representing the dataset """
+        sample = self.api.create_sample(self.resource_id)
+        if self.api.ok(sample):
+            sample = self.api.get_sample(
+                sample["resource"], "rows=%s" % rows_number)
+            return sample.get("object", {}).get("sample", {}).get("rows")
+        return []
+
+    def get_inputs_sample(self, rows_number=32):
+        """Gets a sample of data representing the orign dataset """
+        if self.origin_dataset is None:
+            raise ValueError("Only datasets that are generated from "
```

Comment:
have you considered the possibility of simply returning a dummy empty sample object? when combining different elements is normally convenient to have a neutral element, like 0 or [], instead of an error... although is true that in the platform we don't have empty datasets, we can have empty samples (e.g. if a filter returns no values).  just a thought, an error might also work well.